operations/helm/charts/mimir-distributed/templates/_helpers.tpl

```diff
@@ -74,6 +74,18 @@ Create the name of the service account
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create the name of the ruler service account
+*/}}
+{{- define "mimir.ruler.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+{{- $sa := default (include "mimir.fullname" .) .Values.serviceAccount.name }}
+{{- printf "%s-%s" $sa "ruler" }}
+{{- else -}}
+    {{ default (include "mimir.serviceAccountName" .) .Values.serviceAccount.name }}
```

Comment:
let's just fall back to the `"mimir.serviceAccountName"` without adding additional default logic. So something like this (haven't tested it)
```suggestion
    {{ include "mimir.serviceAccountName" . }}
```

Reply:
Fine with me.